lib/testrail.rb

```diff
@@ -7,79 +7,102 @@ module Gridium
+					@run_info[:desc] = desc
         end
+        r = _send_request('POST', "#{@url}add_run/#{@pid}", @run_info)
+				if r.key?('error') || r["id"].nil?
+					@run_info[:error] = true
+				else
+					@run_info[:id] = r["id"]
+					Log.debug("[GRIDIUM::TestRail] Run Added: #{r}")
+				end
       end
+			return @run_info[:id]
     end
 
+		# Adds determines what the result of the test is and adds information to various arrays for processing during closing.
+		#
+		# @param rspec_test [RSpec::Example] the example provided by RSpec
+		# @return [bool] determine if case was added or not
 		def add_case(rspec_test)
+			added = false
 			if Gridium.config.testrail
 				Log.debug("[GRIDIUM::TestRail] Adding to list of TestRail Cases...")
 				if rspec_test.nil? then
 					Log.error("[GRIDIUM::TestRail] No test added to results. Turn of Gridium.config.testrail\n")
 				end
 				if rspec_test.exception
-					status = FAILED
+					status = @config[:fail]
 					message = rspec_test.exception.message
 				else
-					status = PASSED
+					status = @config[:pass]
 					message = 'Test Passed.'
 				end
-				test_info = {:trid => rspec_test.metadata[:testrail_id], :status_id => status, :message => message}
-				@testcase_infos.push(test_info)
-				@testcase_ids.push(test_info[:trid])
+				test_info = {:case_id => rspec_test.metadata[:testrail_id], :status_id => status, :comment => message}
+				@tc_results.push(test_info)
+				@tc_ids.push(test_info[:case_id])
+				added = true
 			end
+			return added
 		end
 
+		# Updates the existing test run with test cases and results.  Adds error text for missing test cases if needed. Closes the run as long as it exists.
+		#
+		# @return [bool] if the run was closed or not
 		def close_run
-			if Gridium.config.testrail
-				Log.debug("[GRIDIUM::TestRail] Closing test runid: #{@runid}\n")
-				unless @runid.nil?
-					r = _send_request('POST', "update_run/#{@runid}", {:case_ids => @testcase_ids})
-					@testcase_infos.each do |tc|
-						r = _send_request(
-							'POST',
-							"add_result_for_case/#{@runid}/#{tc[:trid]}",
-							status_id: tc[:status_id],
-							comment: tc[:message]
-							)
-							sleep(0.25)
-					end
-					r = _send_request('POST', "close_run/#{@runid}", nil)
+			closed = false
+			if Gridium.config.testrail && !@run_info[:error]
+				Log.debug("[GRIDIUM::TestRail] Closing test runid: #{@run_info[:id]}\n")
+				r = _send_request('POST', "#{@url}update_run/#{@run_info[:id]}", {:case_ids => @tc_ids})
+				Log.debug("[GRIDIUM::TestRail] UPDATE RUN: #{r}")
+				sleep 0.25
+				r = _send_request('POST', "#{@url}add_results_for_cases/#{@run_info[:id]}", {results: @tc_results})
+				Log.debug("[GRIDIUM::TestRail] ADD RESULTS: #{r}")
+				sleep 0.25
+				Log.debug("#{r.class}")
+				if r.is_a?(Hash)
+					r = _send_request('POST', "#{@url}update_run/#{@run_info[:id]}", {:name => "ER:#{@run_info[:name]}", :description => "#{@run_info[:desc]}\nThe following was returned when adding cases: #{r}"})
```

Comment:
not a big deal, but it's nice to have the payload in a defined variable to make the line easier to read when it is has this many characters